py3status/modules/coin_market.py

```diff
@@ -0,0 +1,223 @@
+
+Examples:
+```
+# view coins in GBP and optionally USD
+coin_market {
+    format_coin = '{name} £{price_gbp:.2f} ${price_usd:.2f} {percent_change_24h}'
+}
+```
+
+@author lasers, x86kernel
+
+SAMPLE OUTPUT
+[
+    {'color': '#FFFF00', 'full_text': 'Bitcoin '},
+    {'color': '#FFFFFF', 'full_text': '$2735.77 '},
+    {'color': '#00FF00', 'full_text': '+2.27%'},
+]
+
+losers
+[
+    {'color': '#FFFF00', 'full_text': 'Bitcoin '},
+    {'color': '#FFFFFF', 'full_text': '$2701.70 '},
+    {'color': '#FF0000', 'full_text': '-0.42%'},
+]
+"""
+
+
+class Py3status:
+    cache_timeout = 600
+    format = '{format_coin}'
+    format_coin = '[\?color=#ff0 {name}] ${price_usd:.2f} {percent_change_24h}'
+    format_separator = ' '
+    markets = ['btc']
+    request_timeout = 5
+    thresholds = [(-100, 'bad'), (0, 'good')]
+
+    def post_config_hook(self):
+        self.first_run = True
+        self.first_use = True
+        self.convert = None
+        self.url = self.reset_url = 'https://api.coinmarketcap.com/v1/ticker/'
+        self.limit = None
+
+        # ranked markets or cherrypicked markets?
+        if isinstance(self.markets, int):
+            self.limit = self.markets
+        else:
+            self.markets = [x.upper().strip() for x in self.markets]
+
+        # create '?convert'
+        for item in self.py3.get_placeholders_list(self.format_coin):
+            if (('price' in item and 'price_btc' not in item) or
+                    '24h_volume' in item or 'market_cap' in item) \
+                    and 'usd' not in item:
+                self.convert = '?convert=%s' % (item.split('_')[-1])
+                self.url = self.reset_url = self.reset_url + self.convert
+                break
+
+        # create '(?|&)limit'
+        if self.limit:
+            self._update_limit(None)
+
+    def _get_coin_data(self, reset_url=False):
+        if reset_url:
+            self.url = self.reset_url
+        try:
+            data = self.py3.request(self.url, timeout=self.request_timeout).json()
+        except self.py3.RequestException:
+            data = {}
+        return data
+
+    def _update_limit(self, data):
+        # we use limit if it exists. otherwise, we stretch the limit
+        # large enough to contain the markets + some padding
+        self.url = self.url + ('&' if self.convert else '?')
+        if self.limit:
+            limit = self.limit
+        else:
+            limit = 0
+            for market_id in self.markets:
+                index = next((i for (i, d) in enumerate(
+                    data) if d['symbol'] == market_id), -1)
+                if index >= limit:
+                    limit = index
+                    limit += 5  # padding
+
+        self.url += 'limit=%s' % limit
+
+    def _strip_data(self, data):
+        # we don't strip if self.limit. otherwise, we reduce 997 coins
+        # to 5 coins by stripping everything not in the markets.
```

Comment:
this comment is very specific about numbers but does not seem to limit to 5 coins